lib/pbench/agent/tool_data_sink.py

```diff
@@ -2060,7 +2073,7 @@ def start(prog: Path, parsed: Arguments):
         # E.g. params = '{ "channel_prefix": "some-prefix",
         #                  "benchmark_run_dir": "/loo/goo" }'
         params = json.loads(params_str)
-        ToolDataSink.fetch_params(params, pbench_run)
+        ToolDataSink.fetch_params(params, pbench_run, parsed.instance_uuid)
```

Comment:
What does this line accomplish?  (Doesn't `fetch_params()` return a value...which we're ignoring here...so what side-effect are we depending on?)

Reply:
Validates the parameters are what we expect before we go through the hassle of daemonizing and instantiating a full blown instance of the Tool Data Sink.  This is existing code behavior.  PRs to suggest a change to the way this work are welcome!

Reply:
It seems that we've blended some concerns which perhaps should be separated:
- here, we want to be able to validate the `params` mapping
- in the other call location, we want to allocate and initialize a `ToolDataSinkParams` (perhaps after validation).

The fact that much of the validation is done simply by accessing the `params` mapping doesn't change this.  And, that quirk of the implementation is not a reason to add a bunch of stuff to the allocation function which should be packaged separately.  (Arguably, the allocation function should be taking a `BenchmarkRunDir` instead of `pbench_run`; or, the caller should be adding it to the returned object.)

Would it be better if we converted the `params` mapping to a `ToolDataSinkParams` object up front, and then passed the object along, so it doesn't have to be (re)created later?

Reply:
But is this PR the place to do that kind of refactoring?

These changes are very simple to follow and understand, albeit not ideal.  Perhaps a follow-on commit could correct that?  Posted PRs are always welcome!